go/service/teams.go

```diff
@@ -249,17 +249,38 @@ func (h *TeamsHandler) GetTeamRootID(ctx context.Context, id keybase1.TeamID) (k
 	return teams.GetRootID(ctx, h.G().ExternalG(), id)
 }
 
-func (h *TeamsHandler) LookupImplicitTeam(ctx context.Context, arg keybase1.LookupImplicitTeamArg) (res keybase1.TeamID, err error) {
+func (h *TeamsHandler) prependUsername(ctx context.Context, name string, public bool) (string, error) {
+	if !public {
+		// Prepend our username on here just in case for private imp teams
+		username := h.G().Env.GetUsername()
+		if len(username) == 0 {
+			return "", libkb.LoginRequiredError{}
+		}
+		return username.String() + "," + name, nil
+	}
+	return name, nil
+}
+
+func (h *TeamsHandler) LookupImplicitTeam(ctx context.Context, arg keybase1.LookupImplicitTeamArg) (res keybase1.LookupImplicitTeamRes, err error) {
 	defer h.G().CTraceTimed(ctx, fmt.Sprintf("LookupImplicitTeam(%s)", arg.Name), func() error { return err })()
-	teamID, _, err := teams.LookupImplicitTeam(ctx, h.G().ExternalG(), arg.Name, arg.Public)
-	return teamID, err
+
+	if arg.Name, err = h.prependUsername(ctx, arg.Name, arg.Public); err != nil {
```

Comment:
Why is this necessary? It doesn't make sense for teams you are a reader of like "other#you".

Reply:
Hmm true, I can move this up a level.

Reply:
same problem exists in kbfs chat i believe

Reply:
At this level it would affect actual KBFS though.